Assets/QA/Tests/XRDeviceActions/_SharedScripts/QuaternionControlActionStatus.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+using UnityEngine.Experimental.Input;
+using UnityEngine.Experimental.Input.Controls;
+using UnityEngine.UI;
+
+public class QuaternionControlActionStatus : MonoBehaviour {
+
+	public InputAction quaternionAction;
+
+    public Text statusText;
+
+	// Use this for initialization
+	void Start () {
+        quaternionAction.Enable();
+        quaternionAction.performed += UpdateQuaternion;
+        quaternionAction.started += UpdateQuaternion;
+        quaternionAction.cancelled += UpdateQuaternion;
+	}
+
+    private void UpdateQuaternion(InputAction.CallbackContext context)
+    {
+        Quaternion value = ((QuaternionControl)(context.control)).ReadValue();
+        statusText.text = QuaternionToFieldText(value);
+    }
+
+    private string QuaternionToFieldText(Quaternion inQuat)
```

Comment:
Can you use https://docs.unity3d.com/ScriptReference/Quaternion.ToString.html to do the formatting?

Reply:
I chose to do it this way, and with a monospace font so that it is super easy to read. Using the other method makes UI spacing difficult and can result in ugly and difficult to decipher results.